presto-tests/src/test/java/io/prestosql/tests/TestMetadataManager.java

```diff
@@ -151,4 +152,16 @@ public void testUpperCaseSchemaIsChangedToLowerCase()
                             return null;
                         });
     }
+
+    @Test
+    public void testUpperCaseListTablesFilter()
+    {
+        assertThat(queryRunner.execute("SELECT * FROM system.jdbc.tables WHERE TABLE_SCHEM = 'upper_case_schema' AND TABLE_NAME = 'upper_case_table'"))
+                .hasSize(1);
+        assertThat(queryRunner.execute("SELECT * FROM system.jdbc.tables WHERE TABLE_SCHEM = 'UPPER_CASE_SCHEMA'"))
+                .isEmpty();
+        assertThat(queryRunner.execute("SELECT * FROM system.jdbc.tables WHERE TABLE_NAME = 'UPPER_CASE_TABLE'"))
+                .isEmpty();
+        // The last 2 queries should return actual tables once https://github.com/prestosql/presto/issues/17 is fixed
```

Comment:
comments like "The last 2 queries" can get outdated, out-placed easily.

Please adorn each test case with a comment separately, eg as I suggested above.

```suggestion
```